drivers/gnss/gnss_u_blox_f9p.c

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT u_blox_f9p
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/gnss.h>
+#include <zephyr/drivers/gnss/gnss_publish.h>
+
+#include <zephyr/modem/ubx.h>
+#include <zephyr/modem/ubx/protocol.h>
+#include <zephyr/modem/ubx/keys.h>
+#include <zephyr/modem/backend/uart.h>
+
+#include "gnss_ubx_common.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ubx_f9p, CONFIG_GNSS_LOG_LEVEL);
+
+struct ubx_f9p_config {
+	const struct device *bus;
+	uint16_t fix_rate_ms;
+	uint32_t unused;
+};
+
+struct ubx_f9p_data {
+	struct gnss_ubx_common_data common_data;
+	struct {
+		struct modem_pipe *pipe;
+		struct modem_backend_uart uart_backend;
+		uint8_t receive_buf[1024];
+		uint8_t transmit_buf[256];
+	} backend;
+	struct {
+		struct modem_ubx inst;
+		uint8_t receive_buf[1024];
+	} ubx;
+	struct {
+		struct modem_ubx_script inst;
+		uint8_t response_buf[512];
+		uint8_t request_buf[256];
+		struct k_mutex lock;
+	} script;
+#if CONFIG_GNSS_SATELLITES
+	struct gnss_satellite satellites[CONFIG_GNSS_U_BLOX_F9P_SATELLITES_COUNT];
+#endif
+};
+
+UBX_FRAME_DEFINE(disable_gga,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GGA_UART1, 0));
+UBX_FRAME_DEFINE(disable_rmc,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_RMC_UART1, 0));
+UBX_FRAME_DEFINE(disable_gsv,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GSV_UART1, 0));
+UBX_FRAME_DEFINE(disable_dtm,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_DTM_UART1, 0));
+UBX_FRAME_DEFINE(disable_gbs,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GBS_UART1, 0));
+UBX_FRAME_DEFINE(disable_gll,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GLL_UART1, 0));
+UBX_FRAME_DEFINE(disable_gns,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GNS_UART1, 0));
+UBX_FRAME_DEFINE(disable_grs,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GRS_UART1, 0));
+UBX_FRAME_DEFINE(disable_gsa,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GSA_UART1, 0));
+UBX_FRAME_DEFINE(disable_gst,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_GST_UART1, 0));
+UBX_FRAME_DEFINE(disable_vlw,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_VLW_UART1, 0));
+UBX_FRAME_DEFINE(disable_vtg,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_VTG_UART1, 0));
+UBX_FRAME_DEFINE(disable_zda,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_NMEA_ZDA_UART1, 0));
+UBX_FRAME_DEFINE(enable_nav,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_MSG_OUT_UBX_NAV_PVT_UART1, 1));
+UBX_FRAME_DEFINE(nav_fix_mode_auto,
+	UBX_FRAME_CFG_VAL_SET_U8_INITIALIZER(UBX_KEY_NAV_CFG_FIX_MODE, UBX_FIX_MODE_AUTO));
```

Comment:
Users might want an easy way to force this to 3D only outside of setting the dyn model.